XenAdmin/Wizards/PatchingWizard/PatchingWizard_AutomatedUpdatesPage.cs

```diff
@@ -130,7 +131,13 @@ protected override void GeneratePlanActions(Pool pool, List<HostPlan> planAction
             }
 
             //add a revert pre-check action for this pool
-            var problemsToRevert = ProblemsResolvedPreCheck.Where(p => hosts.ToList().Select(h => h.uuid).ToList().Contains(p.Check.Host.uuid)).ToList();
+            var problemsToRevert = ProblemsResolvedPreCheck.Where(p =>
+            {
+                var hostCheck = p.Check as HostCheck;
+                if (hostCheck != null)
+                    return hosts.ToList().Select(h => h.uuid).ToList().Contains(hostCheck.Host.uuid);

```

Comment:
`ToList()` is not needed here: `hosts.Select(h => h.uuid).Contains(hostCheck.Host.uuid)`